pkg/controller/containersource/reconcile.go

```diff
@@ -133,29 +135,46 @@ func (r *reconciler) getSinkUri(ctx context.Context, source *v1alpha1.ContainerS
 		return "", fmt.Errorf("sink ref is nil")
 	}
 
-	obj, err := r.fetchObjectReference(source.Namespace, source.Spec.Sink)
+	obj, err := r.fetchObjectReference(ctx, source.Namespace, source.Spec.Sink)
 	if err != nil {
-		logger.Warnf("Failed to fetch sink target %+v: %s", source.Spec.Sink, err)
+		logger.Warnf("Failed to fetch sink target %+v: %s", source.Spec.Sink, zap.Error(err))
 		return "", err
 	}
 	t := duckv1alpha1.Sink{}
 	err = duck.FromUnstructured(obj, &t)
 	if err != nil {
-		logger.Warnf("Failed to deserialize sink: %s", err)
+		logger.Warnf("Failed to deserialize sink: %s", zap.Error(err))
 		return "", err
 	}
 
 	if t.Status.Sinkable != nil {
-		return t.Status.Sinkable.DomainInternal, nil
+		return fmt.Sprintf("http://%s/", t.Status.Sinkable.DomainInternal), nil
+	}
+
+	// for now we will try again as a targetable.
+	if tryTargetable {
+		t := duckv1alpha1.Target{}
+		err = duck.FromUnstructured(obj, &t)
+		if err != nil {
+			logger.Warnf("Failed to deserialize targetable: %s", zap.Error(err))
+			return "", err
+		}
+
+		if t.Status.Targetable != nil {
+			return fmt.Sprintf("http://%s/", t.Status.Targetable.DomainInternal), nil
+		}
 	}
+
 	return "", fmt.Errorf("sink does not contain sinkable")
 }
 
 // fetchObjectReference fetches an object based on ObjectReference.
-func (r *reconciler) fetchObjectReference(namespace string, ref *corev1.ObjectReference) (duck.Marshalable, error) {
+func (r *reconciler) fetchObjectReference(ctx context.Context, namespace string, ref *corev1.ObjectReference) (duck.Marshalable, error) {
+	logger := logging.FromContext(ctx)
+
 	resourceClient, err := r.CreateResourceInterface(namespace, ref)
 	if err != nil {
-		//glog.Warningf("failed to create dynamic client resource: %v", err)
+		logger.Warnf("failed to create dynamic client resource: %v", zap.Error(err))
```

Comment:
Could we just return the error here? The caller also logs errors at https://github.com/knative/eventing-sources/pull/16/files#diff-28bc3fe544a2ddf18f611c0b8e4ac2c8R138. That also eliminates the ctx arg.